src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const results = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        if (action.keysToRemove && action.keysToRemove.length > 0) {
+          for (const key of action.keysToRemove) {
+            if (currentState.hasOwnProperty(key)) {
+              delete currentState[key];
+            }
+          }
+        }
+        break;
+      default:
+        continue;
```

Comment:
The `default` case in the switch statement should handle unexpected action types. Using `continue` here might not be the best approach since it assumes the loop will continue executing. It's better to throw an error or handle the unexpected case explicitly.

Reply:
You have to fix this comment